content/essentials/refs.mdx

```diff
@@ -0,0 +1,106 @@
+# Refs
+
+Refs, or references, allow you to have direct access to DOM element. 
+
+To create a `ref` in Solid, you can get a reference to an element through assigning it to avariable: 
+
+```jsx
+const myDiv = <div>My Element</div>;
+```
+
+For a more idiomatic way, Solid provides the `ref` attribute. 
+Refs in Solid are assignments that occur at creation time before elements are attached to the document DOM.
+To use this, you simply need to declare a variable and use it as a ref attribute:
+
+```tsx
+let myDiv;
+
+<div ref={myDiv}>My Element</div>
+
+```
+
+This lets you access its properties and methods directly, allowing for manipulatinos such as animations and event handling.
+
```

Comment:
Would love a full component with an onMount(() => {console.log(ref)}) here

Reply:
could you please write this example out and we can put it in?

Reply:
```ts
const MyComponent = () => {
  let myDiv!: HTMLDivElement;

  onMount(() => {console.log(ref)});

  return <div ref={myDiv}>My Element</div>
}
```